core/src/main/java/org/elasticsearch/ingest/core/CompoundProcessor.java

```diff
@@ -82,19 +82,28 @@ public void execute(IngestDocument ingestDocument) throws Exception {
         }
     }
 
-    void executeOnFailure(IngestDocument ingestDocument, Exception cause, String failedProcessorType, String failedProcessorTag) throws Exception {
+    public void putFailureMetadata(IngestDocument ingestDocument, Exception cause, String failedProcessorType, String failedProcessorTag) {
```

Comment:
I do not feel comfortable making this public, but I did not see another way around this without more large-scale refactoring